lib/components/desktop/wallpaper_picker.dart

```diff
@@ -97,58 +99,80 @@ class _WallpaperPickerState extends State<WallpaperPicker>
                   //
                   //Default Wallpapers
                   //
-                  GridView.builder(
-                    physics: const BouncingScrollPhysics(),
-                    gridDelegate:
-                        const SliverGridDelegateWithFixedCrossAxisCount(
-                      crossAxisCount: 4,
-                      childAspectRatio: 16 / 9,
-                    ),
-                    itemCount: wallpapers.length,
-                    itemBuilder: (BuildContext context, int index) {
-                      //_index = index;
-                      return Padding(
-                        padding: const EdgeInsets.all(8.0),
-                        child: InkWell(
-                          onTap: () {
-                            service.wallpaper = builtinWalls[index];
-                            service.addRecentWallpaper(service.wallpaper);
-                          },
-                          child: Stack(
-                            children: [
-                              Positioned.fill(
-                                child: ResourceImage(
-                                  resource: builtinWalls[index],
-                                  fit: BoxFit.cover,
+                  FutureBuilder<List<Wallpaper?>?>(
+                    future: wallpapers,
+                    builder: (
+                      BuildContext context,
+                      AsyncSnapshot<List<Wallpaper?>?> snapshot,
+                    ) {
+                      if (snapshot.hasData) {
```

Comment:
use shortcircuits instead of a huge if/else if/else block as it's easier to maintain and read.

example:

```dart
if(snapshot.data == null) {
  return LoadingIndicator(); //or anything else really
}

if(snapshot.hasError) {
  return ErrorWidget();
}

return ...
```